packages/pyright-internal/src/languageService/signatureHelpProvider.ts

```diff
@@ -141,13 +143,28 @@ export class SignatureHelpProvider {
                 paramName = params[params.length - 1].name || '';
             }
 
+            if (paramName && !paramString.includes(paramName) && paramName === 'kwargs') {
+                // Param string may have been expanded from kwargs.
+                // Pull out the param name from the string.
+                const match = /^(\w+):/.exec(paramString);
```

Comment:
if type printer can return some kind of structured data rather than plain string to avoid doing this kind of thing would be nice. but I guess that's something we would do in different PR.

Reply:
Yeah, this is pretty hack. I recommend finding a cleaner way to do this.